internal/broker/broker-operation.go

```diff
@@ -70,45 +70,72 @@ func (operation *Operation) AlterBroker(id string, flags AlterBrokerFlags) error
 		return errors.Errorf("cannot find broker with id: %s", id)
 	}
 
-	var configs []internal.Config
+	var existingConfigs []sarama.ConfigEntry
 	brokerConfig := sarama.ConfigResource{
 		Type: sarama.BrokerResource,
 		Name: id,
 	}
 
-	if configs, err = internal.ListConfigs(&admin, brokerConfig, false); err != nil {
-		return err
+	if existingConfigs, err = admin.DescribeConfig(brokerConfig); err != nil {
+		return errors.Wrap(err, "failed to describe config for broker")
 	}
 
-	if len(flags.Configs) > 0 {
-		mergedConfigEntries := make(map[string]*string)
+	mergedConfigEntries := make(map[string]*string)
+	for _, existingConfig := range existingConfigs {
+		if !existingConfig.ReadOnly && !existingConfig.Default {
+			mergedConfigEntries[existingConfig.Name] = &(existingConfig.Value)
```

Comment:
The problem is that when you get all the configs and then try to send them back, you might get an error if the config is not a dynamic config (let's say you use "" as the broker id).

Reply:
Thats why I filter `!existingConfig.ReadOnly`. This should filter out the non dynamic configs